pins/constructors.py

```diff
@@ -511,19 +513,36 @@ def board_s3(path, versioned=True, cache=DEFAULT, allow_pickle_read=None):
         You can enable reading pickles by setting this to `True`, or by setting the
         environment variable `PINS_ALLOW_PICKLE_READ`. If both are set, this argument
         takes precedence.
+    storage_options:
+        Additional keyword arguments to be passed to the underlying fsspec S3FileSystem.
 
     Notes
     -----
     The s3 board uses the fsspec library (s3fs) to handle interacting with AWS S3.
     In order to authenticate, set the `AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY`,
-    and (optionally) `AWS_REGION` environment variables.
+    and (optionally) `AWS_REGION` environment variables. If you are using an
+    s3-compatible storage service that is not from AWS, you can pass in the necessary
+    credentials to the `storage_options` dictionary, such as `endpoint_url`, `key`, and
+    `secret`. We recommend setting these as environment variables. An example using
+    Backblaze B2 would look like:
+
+    Examples
+    --------
+    >>> import pins
+    >>> board = pins.board_s3(
+    ...     "pins-test",
+    ...     endpoint_url=os.getenv("FSSPEC_S3_ENDPOINT_URL"),
+    ...     key=os.getenv("FSSPEC_S3_KEY"),
+    ...     secret=os.getenv("FSSPEC_S3_SECRET"),
+    ... )
 
     See <https://github.com/fsspec/s3fs>
 
     """
     # TODO: user should be able to specify storage options here?
 
     opts = {"listings_expiry_time": 0}
+    opts.update(**storage_options)
```

Comment:
We should probably check `storage_options` and raise a warning if people overwrite the `listings_expiry_time`. It's okay to not error/fully block people from overwriting it, but just note there may be surprising behavior on `pin_read` if this value is not 0. 


Info on the [listings cache](https://filesystem-spec.readthedocs.io/en/latest/features.html#listings-caching) that explains what `listings_expiry_time` does, if you're interested.

Reply:
On it, I'll get to it right away.